test/AVCStyleMcpTest.cc

```diff
@@ -0,0 +1,589 @@
+typedef void (*MCP_TEST_FUNC)(EbByte ref_pic, uint32_t src_stride, EbByte dst,
+                              uint32_t dst_stride, uint32_t pu_width,
+                              uint32_t pu_height, EbByte temp_buf, EbBool skip,
+                              uint32_t frac_pos);
+typedef struct {
+    const char *name;
+    MCP_REF_FUNC ref_func;
+    MCP_TEST_FUNC test_func;
+} AVCStyleMcpFuncPair;
+
+static const AVCStyleMcpFuncPair AVC_style_c_sse3_func_pairs[12] = {
+    {"posA", WpAvcStyleCopy, avc_style_copy_sse2},
+    {"pose",
+     WpAvcStyleLumaInterpolationFilterPose,
+     avc_style_luma_interpolation_filter_pose_ssse3},
+    {"posf",
+     WpAvcStyleLumaInterpolationFilterPosf,
+     avc_style_luma_interpolation_filter_posf_ssse3},
+    {"posg",
+     WpAvcStyleLumaInterpolationFilterPosg,
+     avc_style_luma_interpolation_filter_posg_ssse3},
+    {"posi",
+     WpAvcStyleLumaInterpolationFilterPosi,
+     avc_style_luma_interpolation_filter_posi_ssse3},
+    {"posj",
+     WpAvcStyleLumaInterpolationFilterPosj,
+     avc_style_luma_interpolation_filter_posj_ssse3},
+    {"posk",
+     WpAvcStyleLumaInterpolationFilterPosk,
+     avc_style_luma_interpolation_filter_posk_ssse3},
+    {"posp",
+     WpAvcStyleLumaInterpolationFilterPosp,
+     avc_style_luma_interpolation_filter_posp_ssse3},
+    {"posq",
+     WpAvcStyleLumaInterpolationFilterPosq,
+     avc_style_luma_interpolation_filter_posq_ssse3},
+    {"posr",
+     WpAvcStyleLumaInterpolationFilterPosr,
+     avc_style_luma_interpolation_filter_posr_ssse3},
+    {"vertical",
+     WpAvcStyleLumaInterpolationFilterVertical,
+     avc_style_luma_interpolation_filter_vertical_ssse3_intrin},
+    {"horizontal",
+     WpAvcStyleLumaInterpolationFilterHorizontal,
+     avc_style_luma_interpolation_filter_horizontal_ssse3_intrin}};
+
+typedef std::tuple<PUSize, TestPattern> TestPUParam;
+typedef std::tuple<SearchArea, TestPattern> TestSearchRegionParam;
+
+/**
+ * @Brief Base class for Mcp test.AVCStyleMcpTestBase handle
+ *  test vector in memory,provide reference functions.
+ */
+class AVCStyleMcpTestBase : public ::testing::Test {
+  public:
+    AVCStyleMcpTestBase(TestPattern test_pattern) {
+        test_pattern_ = test_pattern;
+        ref_stride_ = MAX_PICTURE_WIDTH_SIZE;
+        tst_stride_ = MAX_PICTURE_WIDTH_SIZE;
+    }
+
+    AVCStyleMcpTestBase(const uint32_t pu_width, const uint32_t pu_height,
+                        EbBool isPU, TestPattern test_pattern) {
+        isPU = isPU;
+        pu_width_ = pu_width;
+        pu_height_ = pu_height;
+        test_pattern_ = test_pattern;
+        tst_size = MAX_PU_SIZE * MAX_PU_SIZE;
+        ref_offset = 2 + 2 * (MAX_PICTURE_WIDTH_SIZE + 64 + 4 + 64 + 4);
+        ref_stride_ = MAX_PICTURE_WIDTH_SIZE + 64 + 4 + 64 + 4;
+        tst_stride_ = 2 * MAX_PU_SIZE;
+        ref_aligned_ = (uint8_t *)eb_aom_memalign(8, MAX_PIC_SIZE);
+        avc_tmp_ = (uint8_t *)eb_aom_memalign(8, 4 * tst_size);
```

Comment:
It will be great if we can add some comments about the size, like avc_tmp_, ref_offset, or ref_stride.